pay-api/src/pay_api/models/routing_slip.py

```diff
@@ -62,6 +69,95 @@ def find_by_number(cls, number: str) -> RoutingSlip:
+                result, count = pagination.items, max_no_records
+            else:
+                result = query.all()
+                count = len(result)
+
+        return result, count
+
+    @classmethod
+    def _add_date_filter(cls, query, search_filter):
+        # Find start and end dates for folio search
+        created_from: datetime = None
+        created_to: datetime = None
+        if get_str_by_path(search_filter, 'dateFilter/endDate'):
+            created_to = datetime.strptime(get_str_by_path(search_filter, 'dateFilter/endDate'), '%m/%d/%Y')
+        if get_str_by_path(search_filter, 'dateFilter/startDate'):
+            created_from = datetime.strptime(get_str_by_path(search_filter, 'dateFilter/startDate'), '%m/%d/%Y')
+        if get_str_by_path(search_filter, 'weekFilter/index'):
+            created_from, created_to = get_week_start_and_end_date(
+                int(get_str_by_path(search_filter, 'weekFilter/index')))
+        if get_str_by_path(search_filter, 'monthFilter/month') and get_str_by_path(search_filter,
+                                                                                   'monthFilter/year'):
+            # find month
+            month = int(get_str_by_path(search_filter, 'monthFilter/month'))
+            year = int(get_str_by_path(search_filter, 'monthFilter/year'))
+            created_from, created_to = get_first_and_last_dates_of_month(month=month, year=year)
+        # if passed in details
+        if created_to and created_from:
+            # Truncate time for from date and add max time for to date
+            tz_name = current_app.config['LEGISLATIVE_TIMEZONE']
+            tz_local = pytz.timezone(tz_name)
+
+            created_from = created_from.replace(hour=0, minute=0, second=0, microsecond=0).astimezone(tz_local)
+            created_to = created_to.replace(hour=23, minute=59, second=59, microsecond=999999).astimezone(tz_local)
+            query = query.filter(
+                func.timezone(tz_name, func.timezone('UTC', RoutingSlip.created_on)).between(created_from, created_to))
+        return query
+
+    @classmethod
+    def _add_receipt_number(cls, query, search_filter):
+        if receipt_number := search_filter.get('receiptNumber', None):
+            query = query.join(Payment).filter(
+                and_(Payment.payment_account_id == PaymentAccount.id, Payment.receipt_number == receipt_number))
+        return query
+
+    @classmethod
+    def _add_folio_filter(cls, query, search_filter):
+        if folio_number := search_filter.get('folioNumber', None):
+            query = query.join(Invoice).filter(
+                and_(Invoice.routing_slip == RoutingSlip.number, and_(Invoice.folio_number == folio_number,
+                                                                      Invoice.payment_method_code.in_(
+                                                                          [PaymentMethod.CASH.value,
```

Comment:
This will change, we may not populate the payment method as CASH or CHEQUE in invoice table, as with linking a Cash routing slip can be linked with Cheque and vice versa. So we won't be able to specify the payment method, instead will populate a generic one. The join already exists in the model, so can't we use it (to avoid changes in multiple places later) ?

Reply:
fixed
But couldnt reuse the existing join